flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/ZooKeeperDefaultDispatcherRunnerTest.java

```diff
@@ -180,7 +179,17 @@ public void testResourceCleanupUnderLeadershipChange() throws Exception {
                     createDispatcherRunner(
                             rpcService,
                             dispatcherLeaderElectionService,
-                            () -> createZooKeeperJobGraphStore(client),
+                            new JobPersistenceComponentFactory() {
```

Comment:
again, `new TestingJobPersistenceComponentFactory(createZooKeeperJobGraphStore(client), new EmbeddedJobResultStore())` might be nice to have